### ISSUE TITLE
Dropwizard BOM provides dependency management for modules outside the Dropwizard project

### ISSUE THREAD

**Issue by srdo** [dropwizard/dropwizard#2880]
The BOM is setting dependency management for modules that aren't part of Dropwizard. This is not ideal, as it can cause conflicts when other projects import the BOM.

A simple example would be if I had a project using the Dropwizard BOM, and also depending on something that depends on `tomcat-jdbc:9.0.8` and `tomcat-dbcp:9.0.8`. In this case I would get `tomcat-jdbc:9.0.22` from Dropwizard, and `tomcat-dbcp:9.0.8`, which might not work together. This will happen even if I don't use a Dropwizard module that requires `tomcat-jdbc`.

See a similar discussion here https://github.com/dropwizard/metrics/issues/1336

Ideally the BOM would only set the versions of io.dropwizard modules, and let the consuming project handle resolving any conflicts between transitive dependencies. This is also how several other modular projects do their BOMs (e.g. JUnit, Jackson, Camel)

**Comment by joschi** [comment]
@dropwizard/committers Any strong opinions on this?

The advantage of having all dependencies in `dropwizard-bom` is that every project automatically can use the pre-configured (transitive) dependencies without having to declare their versions.

In that way it's similar to `org.springframework.boot:spring-boot-dependencies`:
https://github.com/spring-projects/spring-boot/blob/7712f9c0e5a84f04a9c93c195b90d05cfcf50a67/spring-boot-project/spring-boot-dependencies/pom.xml

The difference between JUnit, Jackson, Camel, and Dropwizard (or Spring Boot) is that the latter is a framework that basically rules how to build your application while the others are libraries which are usually added to existing applications.

**Comment by jplock** [comment]
I like having everything self contained

**Comment by srdo** [comment]
Just to make sure we have a common understanding of what removing the transitive dependencies from the BOM would do: Consuming projects get Dropwizard's transitive dependencies on their classpaths regardless of whether the BOM mentions them or not, the only difference is whether the BOM overrides Maven's normal transitive dependency resolution for these artifacts. 

Let me give some examples using `tomcat-jdbc` (these artifacts don't necessarily make sense together, but illustrate the principle):

Example consumer POM:
```
<dependency>
    <groupId>io.dropwizard</groupId>
    <artifactId>dropwizard-bom</artifactId>
    <type>pom</type>
    <scope>import</scope>
</dependency>
<dependency>
    <groupId>io.dropwizard</groupId>
    <artifactId>dropwizard-core</artifactId>
</dependency>
<dependency>
    <!-- Uses tomcat-jdbc -->
    <groupId>com.zaxxer</groupId>
    <artifactId>HikariCP</artifactId>
</dependency>
```

With the current BOM, the `tomcat-jdbc` version is determined by the BOM. As I'm not using a part of Dropwizard that depends on `tomcat-jdbc`, this is surprising behavior. I can work around this by declaring my own `dependencyManagement` section, which overrides the BOM, but this is unnecessary hassle in this case, since there is no reason for Dropwizard to care which version of `tomcat-jdbc` I'm using here.

With a BOM where transitive dependencies aren't declared, the `tomcat-jdbc` version is  determined by Maven's ordinary dependency resolution. In this case I would get the version pointed to by HikariCP.

Example 2:
```
<dependency>
    <groupId>io.dropwizard</groupId>
    <artifactId>dropwizard-bom</artifactId>
    <type>pom</type>
    <scope>import</scope>
</dependency>
<dependency>
    <groupId>io.dropwizard</groupId>
    <artifactId>dropwizard-core</artifactId>
</dependency>
<dependency>
    <!-- Uses tomcat-jdbc -->
    <groupId>com.zaxxer</groupId>
    <artifactId>HikariCP</artifactId>
</dependency>
<dependency>
    <!-- Uses tomcat-jdbc -->
    <groupId>io.dropwizard</groupId>
    <artifactId>dropwizard-db</artifactId>
</dependency>
```

With the current BOM, I get the version used by Dropwizard. If there's incompatibility with HikariCP, I need to resolve it somehow. 

With a BOM where transitive dependencies aren't declared, Maven would determine the `tomcat-jdbc` version for me. In this case I'd get the Hikari version, as it's declared first. This is exactly the same situation as with the current BOM, since I still need to check that the Dropwizard and Hikari `tomcat-jdbc` versions are compatible.

Example 3:
```
<dependency>
    <groupId>io.dropwizard</groupId>
    <artifactId>dropwizard-bom</artifactId>
    <type>pom</type>
    <scope>import</scope>
</dependency>
<dependency>
    <groupId>io.dropwizard</groupId>
    <artifactId>dropwizard-core</artifactId>
</dependency>
<dependency>
    <!-- Uses tomcat-jdbc -->
    <groupId>io.dropwizard</groupId>
    <artifactId>dropwizard-db</artifactId>
</dependency>
```

With the current BOM, I get the version declared in the BOM.

With a BOM where transitive dependencies aren't declared, I'd get the `tomcat-jdbc` declared in the `dropwizard-db` POM. This is the same version as would be in the BOM.

The only case here where having transitive dependencies in the BOM helps is in example 2, if  the Dropwizard `tomcat-jdbc` version is compatible with the chosen HikariCP version, but not the other way around. Whether this behavior is helpful is up to coincidence, since I could have declared the dependencies in reverse order (Maven would then pick Dropwizard's version) or Hikari's version could be the one that works for both projects.

The downside to declaring transitive dependencies in the BOM is that it expands the number of dependency conflicts the consuming project needs to resolve. If I have a dependency that conflicts with Dropwizard and I'm using that part of Dropwizard, I need to resolve it whether the BOM has transitive declarations or not. When the BOM has transitive declarations, I potentially additionally need to override those declarations, even if I'm not using the part of Dropwizard that has those dependencies.

**Comment by black-snow** [comment]
I don't know if this is related but there is a BOM's pom that's tagged and released as `9.4.20` but mentions jetty-util `9.4.18` and breaks my websockets for `org.eclipse.jetty.util.compression.InflaterPool` is not present in `9.4.18`.
Took me about an hour to figure out where the 9.4.18 util came from.

**Comment by joschi** [comment]
The dependencies have been reorganized in #2897.

@black-snow Could you please create a new issue for this and add a minimal example to reproduce the problem?

**Comment by black-snow** [comment]
Will most likely be fixed with 2.0.0.
